src/haversine.jl

```diff
@@ -11,7 +11,8 @@ The computed distance has the unit of the radius.
 struct Haversine{T<:Number} <: Metric
     radius::T
 end
-Haversine() = Haversine(Float32(6_371_000))
+Haversine{T}() where {T<:Number} = Haversine(T(6_371_000))
+Haversine() = Haversine{Int}()
```

Comment:
```suggestion
Haversine() = Haversine{Int32}()
```
? Otherwise there seems to be no chance for (in the default) case that `Float32` inputs yield `Float32` output.

Reply:
No it does:)

```julia
julia> promote_type(Int64, Float32)
Float32
```

Reply:
Excellent. I should stop guessing and test things on my own. :-))))) A test would be good then.

Reply:
It's really funny that we didn't think about it at https://github.com/JuliaStats/Distances.jl/issues/216. :-D